utilities/test_suite/HIP/runAudioTests.py

```diff
@@ -0,0 +1,295 @@
+    parser.add_argument('--qa_mode', type = int, default = 0, help = "Run with qa_mode? Output audio data from tests will be compared with golden outputs - (0 / 1)", required = False)
+    parser.add_argument('--case_list', nargs = "+", help = "List of case numbers to test", required = False)
+    parser.add_argument('--profiling', type = str , default = 'NO', help = 'Run with profiler? - (YES/NO)', required = False)
+    parser.add_argument('--num_runs', type = int, default = 1, help = "Specifies the number of runs for running the performance tests")
+    parser.add_argument('--preserve_output', type = int, default = 1, help = "preserves the output of the program - (0 = override output / 1 = preserve output )")
+    parser.add_argument('--batch_size', type = int, default = 1, help = "Specifies the batch size to use for running tests. Default is 1.")
+    args = parser.parse_args()
+
+    # check if the folder exists
+    validate_path(args.input_path)
+
+    # validate the parameters passed by user
+    if ((args.case_start < caseMin or args.case_start > caseMax) or (args.case_end < caseMin or args.case_end > caseMax)):
+        print(f"Starting case# and Ending case# must be in the {caseMin}:{caseMax} range. Aborting!")
+        exit(0)
+    elif args.case_end < args.case_start:
+        print("Ending case# must be greater than starting case#. Aborting!")
+        exit(0)
+    elif args.test_type < 0 or args.test_type > 1:
+        print("Test Type# must be in the 0 / 1. Aborting!")
+        exit(0)
+    elif args.case_list is not None and args.case_start != caseMin and args.case_end != caseMax:
+        print("Invalid input! Please provide only 1 option between case_list, case_start and case_end")
+        exit(0)
+    elif args.qa_mode < 0 or args.qa_mode > 1:
+        print("QA mode must be in the 0 / 1. Aborting!")
+        exit(0)
+    elif args.num_runs <= 0:
+        print("Number of Runs must be greater than 0. Aborting!")
+        exit(0)
+    elif args.batch_size <= 0:
+        print("Batch size must be greater than 0. Aborting!")
+        exit(0)
+    elif args.profiling != 'YES' and args.profiling != 'NO':
+        print("Profiling option value must be either 'YES' or 'NO'.")
+        exit(0)
+    elif args.preserve_output < 0 or args.preserve_output > 1:
+        print("Preserve Output must be in the 0/1 (0 = override / 1 = preserve). Aborting")
+        exit(0)
+    elif args.test_type == 0 and args.input_path != inFilePath:
+        print("Invalid input path! QA mode can run only with path:", inFilePath)
+        exit(0)
+
+    if args.case_list is None:
+        args.case_list = range(args.case_start, args.case_end + 1)
+        args.case_list = [str(x) for x in args.case_list]
+    else:
+        for case in args.case_list:
+            if int(case) < caseMin or int(case) > caseMax:
+                print(f"Invalid case number {case}! Case number must be in the {caseMin}:{caseMax} range. Aborting!")
```

Comment:
I think we are using fstring here, which might create problem
Please check and change this

Reply:
Done